src/ln/channel.rs

```diff
@@ -239,18 +240,24 @@ enum ChannelState {
 }
 const BOTH_SIDES_SHUTDOWN_MASK: u32 = (ChannelState::LocalShutdownSent as u32 | ChannelState::RemoteShutdownSent as u32);
 
+
+
+
 // TODO: We should refactor this to be an Inbound/OutboundChannel until initial setup handshaking
 // has been completed, and then turn into a Channel to get compiler-time enforcement of things like
 // calling channel_id() before we're set up or things like get_outbound_funding_signed on an
 // inbound channel.
 pub struct Channel {
+	///TODO: split limits and runtime setting here
+	/// channel limits cant change during channel existence
+	/// but other settings can, but they need an update broadcast before they can change. 
+	config : UserConfigurations, 
 	user_id: u64,
 
 	channel_id: [u8; 32],
 	channel_state: u32,
 	channel_outbound: bool,
 	secp_ctx: Secp256k1<secp256k1::All>,
-	announce_publicly: bool,
```

Comment:
I dont think you can remove this? For outbound channels it will always match our config announce_channel (which was renamed from require_announce), but for *inbound* channels the semantics are still that of require_announce - if we don't set require_announce and set allow_announce the remote side can set announcement to anything it likes (and we need to keep track of what it is and it cannot change for the duration of a channel).

Reply:
thats saved in the config. Config now has require and allow announce. By locking the config we keep this for the duration of the config?

Reply:
Hmm? If the config is set as channel_options.announce_channel = false (which is confusingly named) and channel_options.allow_announce_channel = true then we'll accept the OpenChannel either way. However, we need to store whether the OpenChannel flags were set to announce or not, as we're required to either follow what they said or reject the channel.